vspace/src/main/java/edu/asu/diging/vspace/web/staff/AddSpaceController.java

```diff
@@ -47,4 +48,19 @@ public String addSpace(Model model, @ModelAttribute SpaceForm spaceForm, @Reques
 
 		return "redirect:/staff/space/list";
 	}
+	
+	@RequestMapping(value = "/staff/space/update/{id}", method = RequestMethod.POST)
+	public String updateSpace(@PathVariable("id") String id, Model model, @RequestParam("file") MultipartFile file, Principal principal) throws IOException {
+
+		byte[] bgImage = null;
+		String filename = null;
+		if (file  != null) {
```

Comment:
what if file is null?

Reply:
File cant be null, as we alert the user to select the file before hitting change Image button.